sycl/include/sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp

```diff
@@ -29,6 +30,46 @@ namespace {
     return *this;                                                              \
   }
 
+// compare strings on compiletime
+template <const char *const &str1, const char *const &str2> struct compareStrs {
```

Comment:
Can it be shortened? E.g., see https://stackoverflow.com/questions/27490858/how-can-you-compare-two-character-strings-statically-at-compile-time